### ISSUE TITLE
Unique naming to columns

### ISSUE THREAD

**Issue by piever** [JuliaPlots/AlgebraOfGraphics.jl#69]
In the pair syntax (e.g. `:a => categorical`), give names that take the function into account (as does DataFrames).

**Comment by greimel** [comment]
... and (if that doesn't work yet) also allow renaming?

**Comment by piever** [comment]
Yes, I guess we should basically copy DataFrames, so basically `:col => func` gives you `df.col_func = func(df.col)`, and `:col => func => :new_name` gives `df.new_name = func(df.col)`. Maybe even `:col => :newname`, but let's wait to see if its' actually needed.

**Comment by jtrakk** [comment]
I like "mean(val)" better than val_mean.
